Formula/elan-init.rb

```diff
@@ -0,0 +1,54 @@
+class ElanInit < Formula
+  desc "Lean Theorem Prover installer and version manager"
+  homepage "https://leanprover-community.github.io/"
+  url "https://github.com/leanprover/elan/archive/v1.0.3.tar.gz"
+  sha256 "fb3ddd8915a0694ead0f3a51fdf8b0a5540f983f44eee0e757339244c522b8ee"
+  license "Apache-2.0"
+  head "https://github.com/leanprover/elan.git"
+
+  depends_on "rust" => :build
+
+  # elan-init will run on arm64 Macs, but will fetch Leans that are x86_64.
+  depends_on arch: :x86_64
+
+  def install
+    cargo_home = buildpath/"cargo_home"
+    cargo_home.mkpath
+    ENV["CARGO_HOME"] = cargo_home
+    ENV["RELEASE_TARGET_NAME"] = "homebrew-build"
+
+    system "cargo", "install", "--features", "no-self-update", *std_cargo_args
+
+    mv bin/"elan-init", bin/"elan"
+    %w[lean leanpkg leanchecker leanc leanmake].each do |link|
+      ln_s bin/"elan", bin/link
+    end
```

Comment:
```suggestion
    %w[lean leanpkg leanchecker leanc leanmake elan-init].each do |link|
      bin.install_symlink "elan-init" => link
    end
```
Why do we need to install all these symlinks manually?

Reply:
Thanks! Changing.

The reason for the manual symlinking is because users who install this basically want a working `lean` binary (and the other ones) -- the way they get that here is that those symlinks will then make elan do the right thing essentially, so someone can just start using `lean` immediately after install.

This logic is copied from the way this is done in other packaging of elan, namely nix:

https://github.com/NixOS/nixpkgs/blob/838e396944e3d9fe16faf155c6bf869f85f924d0/pkgs/applications/science/logic/elan/default.nix#L45-L61

Reply:
If I follow your guide [1], when do these symlinks get created?

[1] https://leanprover-community.github.io/install/macos_details.html

Reply:
Doing things that way basically is like immediately invoking the binary here during the install. At that point the user gets a prompt, and a default toolchain gets installed.

The way I did it here (which I was pointed to by the Lean dev who maintains the nix expression) essentially delays that install until the first time someone invokes a lean-related command, rather than pre-installing at the moment elan itself is installed, if that makes sense.